### ISSUE TITLE
createLayer error -> exception doesn't have a .message property

### ISSUE THREAD

**Issue by voltagex** [CartoDB/carto.js#1443]
TL;DR: If you pass a bad layer definition to cartodb.createLayer, the error message you get isn't what it should be

**Expected**: `cartodb-leaflet needs at least a CartoDB table name, user_name and tile_style`
**Actual**: `"MAP: error creating 'cartodb' layer -> undefined"`

Bear with me, I'll try to work backwards to the bug
See http://codepen.io/voltagex/pen/KgRbEk

Take the following invalid layer definition:

```
{
   kind: 'cartodb'
}
```

in `cdb.geo.LeafletMapView.cdb.geo.MapView.createLayer`, `new layerClass` is called, passing in the `layer` provided by the user. This is in a `try` block, which fails of course.

https://github.com/CartoDB/cartodb.js/blob/1e770ede4d2764181104322d4d7e5ba9ce393455/src/geo/leaflet/leaflet.js#L377

https://github.com/CartoDB/cartodb.js/blob/1e770ede4d2764181104322d4d7e5ba9ce393455/src/geo/gmaps/gmaps.js#L158

The problem is the cdb.log.error call tries to access `e.message`, which doesn't exist in `e`, so you get an error in the console of `undefined` instead of the much nicer error from https://github.com/CartoDB/cartodb.js/blob/1e770ede4d2764181104322d4d7e5ba9ce393455/src/geo/leaflet/leaflet_cartodb_layer.js#L25 of `cartodb-leaflet needs at least a CartoDB table name, user_name and tile_style`

I hope this makes sense. 

The fix: change https://github.com/CartoDB/cartodb.js/blob/1e770ede4d2764181104322d4d7e5ba9ce393455/src/geo/leaflet/leaflet_cartodb_layer.js#L25 (and the one in the gmaps equivalent) to something like 

``` js
throw ({'message': {'cartodb-leaflet needs at least a CartoDB table name, user_name and tile_style'}});
```
